infraagent/agent/cmd/agent/rootcmd.go

```diff
@@ -87,6 +92,12 @@ func init() {
 	rootCmd.PersistentFlags().StringVar(&config.interfaceName, "interface", "", intfFlagHelpMsg)
 	rootCmd.PersistentFlags().StringVar(&config.cfgFile, "config", "/etc/infra/infraagent.yaml", "config file")
 	rootCmd.PersistentFlags().StringVar(&config.tapPrefix, "tapPrefix", types.TapInterfacePrefix, "Host TAP interface prefix for TAP interface type")
+	rootCmd.PersistentFlags().BoolVar(&config.insecure, "insecure", false, "use insecure mode for internal communication with backend")
+	rootCmd.PersistentFlags().BoolVar(&config.mtls, "mtls", true, "use mTLS for internal communication with backend")
+	rootCmd.PersistentFlags().StringVar(&config.clientCert, "client-cert", types.AgentDefaultClientCert, "TLS Client cert file for mTLS")
+	rootCmd.PersistentFlags().StringVar(&config.clientKey, "client-key", types.AgentDefaultClientKey, "TLS Client key file for mTLS")
+	rootCmd.PersistentFlags().StringVar(&config.caCert, "ca-cert", types.AgentDefaultCACert, "TLS Client CA Cert file")
+
```

Comment:
I guess we also need the server CA to verify the server's certificate. Do we need client CA? It is required at the server end right?

Reply:
So, Agent will not run  it's server with TLS as calico does not support it.
In Infra Manager, yes we will need both client-cert, client-key, server-cert, server-key and ca-cert.
the client-cert, client-key is for communicating to P4d and the server-cert, server-key for exposing service to Infra-Agent.
The ca-cert must be issued by same issuer for Infra agent and server.